File: Homework/Homework 1/C#/Esercise 1/Esercise1A/Form1.cs
using System;
using System.Drawing;
using System.Windows.Forms;

namespace Esercise1A
{
    public partial class Form1 : Form
    {
        
        public Form1()
        {
            InitializeComponent();
        }


        //Click on the button to insert the text
        private void button1_MouseClick(object sender, MouseEventArgs e)
        {
            //Initialize a variable of type String
            String text = "Hi Tommaso Gastaldi";

            //Set the text of the richtextbox element with the contents of the variable text
            this.richTextBox1.Text = text;
        }

        //Click on the button to clean the text
        private void button2_MouseClick(object sender, MouseEventArgs e)
        {
            //The element is completely cleaned and is not set with " " (null)
            this.richTextBox1.Clear();
        }


        //Mouse access in the richtextbox element
        private void richTextBox1_MouseEnter(object sender, EventArgs e)
        {
            //Changing the background color from default to orange
            this.richTextBox1.BackColor = Color.Orange;

            //Changing the text color from default to white
            this.richTextBox1.ForeColor = Color.White;
        }

        //Mouse leave from the richtextbox element
        private void richTextBox1_MouseLeave(object sender, EventArgs e)
        {
            //Changing the background color from orange to default
            this.richTextBox1.BackColor = default(Color);

            //Changing the text color from white to default
            this.richTextBox1.ForeColor = default(Color);
        }
    }
}

File: Homework/Homework 1/VB.NET/Exercise 1/Exercise1/Form1.vb
Public Class Form1

    'Click on the button to insert the text
    Private Sub Button1_MouseClick(sender As Object, e As MouseEventArgs) Handles Button1.MouseClick

        'Initialize a variable of type String
        Dim text As String = "Hi Tommaso Gastaldi!"

        'Set the text of the richtextbox element with the contents of the variable text
        Me.RichTextBox1.Text = text
    End Sub

    'Click on the button to clean the text
    Private Sub Button2_MouseClick(sender As Object, e As MouseEventArgs) Handles Button2.MouseClick

        'The element is completely cleaned and is not set with " " (null)
        Me.RichTextBox1.Clear()
    End Sub

    'Mouse access in the richtextbox element
    Private Sub RichTextBox1_MouseEnter(sender As Object, e As EventArgs) Handles RichTextBox1.MouseEnter

        'Changing the background color from default to orange
        Me.RichTextBox1.BackColor = Color.Orange

        'Changing the text color from default to white
        Me.RichTextBox1.ForeColor = Color.White
    End Sub

    'Mouse leave from the richtextbox element
    Private Sub RichTextBox1_MouseLeave(sender As Object, e As EventArgs) Handles RichTextBox1.MouseLeave

        'Changing the background color from orange to default
        Me.RichTextBox1.BackColor = DefaultBackColor

        'Changing the text color from white to default
        Me.RichTextBox1.ForeColor = DefaultForeColor
    End Sub
End Class
